### PR TITLE
Examples/add google my business location extensions

### PR DESCRIPTION
**Pull request by devchas**

### CODE REVIEW

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/extensions/AddGoogleMyBusinessLocationExtensions.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    // automatically because this will be a system generated feed.
+    Feed gmbFeed =
+        Feed.newBuilder()
+            .setName(StringValue.of("Google My Business feed #" + System.currentTimeMillis()))
+            // Configures the location feed populated from Google My Business Locations.
+            .setPlacesLocationFeedData(
+                PlacesLocationFeedData.newBuilder()
+                    .setEmailAddress(StringValue.of(gmbEmailAddress))
+                    .setBusinessAccountId(StringValue.of(businessAccountIdentifier))
+                    .addLabelFilters(StringValue.of("Stores in New York"))
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/extensions/AddGoogleMyBusinessLocationExtensions.java]: Please add a line comment above this explaining what this filter will do

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/extensions/AddGoogleMyBusinessLocationExtensions.java [hunk 2]
```diff
@@ -123,6 +127,7 @@ public static void main(String[] args) throws InterruptedException, IOException
    * @param gmbAccessToken the access token created using the 'AdWords' scope and the client ID and
    *     client secret of with the cloud project associated with the GMB account.
    * @throws GoogleAdsException if an API request failed with one or more service errors.
+   * @throws InterruptedException if the Thread.sleep operation requires an interrupt.
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/extensions/AddGoogleMyBusinessLocationExtensions.java]: nit: "requires an interrupt" -> "is interrupted"

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/extensions/AddGoogleMyBusinessLocationExtensions.java [hunk 3]
```diff
@@ -142,6 +147,10 @@ private void runExample(
                 PlacesLocationFeedData.newBuilder()
                     .setEmailAddress(StringValue.of(gmbEmailAddress))
                     .setBusinessAccountId(StringValue.of(businessAccountIdentifier))
+                    // Used to filter Google My Business listings by labels. If entries exist in
+                    // label_filters, only listings that has any of the labels set are candidates
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/extensions/AddGoogleMyBusinessLocationExtensions.java]: nit: "only listings that has any of the labels" -> "only listings that **have** at least one of the labels"

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/extensions/AddGoogleMyBusinessLocationExtensions.java [hunk 4]
```diff
@@ -18,30 +18,34 @@
...
-import com.google.ads.googleads.v1.resources.Feed;
-import com.google.ads.googleads.v1.resources.Feed.PlacesLocationFeedData;
-import com.google.ads.googleads.v1.resources.Feed.PlacesLocationFeedData.OAuthInfo;
-import com.google.ads.googleads.v1.services.CustomerFeedOperation;
-import com.google.ads.googleads.v1.services.CustomerFeedServiceClient;
-import com.google.ads.googleads.v1.services.FeedOperation;
-import com.google.ads.googleads.v1.services.FeedServiceClient;
-import com.google.ads.googleads.v1.services.MutateCustomerFeedsResponse;
-import com.google.ads.googleads.v1.services.MutateFeedsResponse;
+import com.google.ads.googleads.v2.common.MatchingFunction;
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/extensions/AddGoogleMyBusinessLocationExtensions.java]: GitHub still shows 5,404 files changed. Is that expected?

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/extensions/AddGoogleMyBusinessLocationExtensions.java [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param gmbEmailAddress email address associated with the GMB account.
+   * @param businessAccountIdentifier the account number of the GMB account.
+   * @param gmbAccessToken the access token created using the 'AdWords' scope and the client ID and
+   *     client secret of with the cloud project associated with the GMB account.
```
Comment (aohren) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/extensions/AddGoogleMyBusinessLocationExtensions.java]: nit: should "cloud" be capitalized?

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/extensions/AddGoogleMyBusinessLocationExtensions.java [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+          numberOfAttempts++;
+          try {
+            MutateCustomerFeedsResponse customerFeedsResponse =
+                customerFeedServiceClient.mutateCustomerFeeds(
+                    Long.toString(customerId), ImmutableList.of(customerFeedOperation));
+            addedCustomerFeed = customerFeedsResponse.getResults(0).getResourceName();
+            System.out.printf("Customer feed created with resource name: %s%n", addedCustomerFeed);
+          } catch (GoogleAdsException gae) {
+            // Waits using exponential backoff policy.
+            long sleepSeconds = (long) Math.scalb(5, numberOfAttempts);
```
Comment (aohren) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/extensions/AddGoogleMyBusinessLocationExtensions.java]: Should we add an upper bound on sleepSeconds? If MAX_CUSTOMER_FEED_ADD_ATTEMPTS is set too high, this sleep will become grow too large.

**Pull request merged**